src/test/java/com/google/sps/CommentRetrievalTest.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.api.services.youtube.model.*;
+import com.google.sps.servlets.utils.YouTubeCommentRetriever;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class CommentRetrievalTest {
+
+  @Before
+  public void setup() {}
+
+  private final int HUNDRED = 100;
+  private final int EXCESS_HUNDRED = 200;
+
+  // A video that has sufficient comments for any test (>900,000).
+  private final String POPULAR_VIDEO_URL = "dQw4w9WgXcQ";
+  // A video with very few comments (Unlikely to ever reach even 100)
+  private final String UNPOPULAR_VIDEO_URL = "cA-arJ0T6L4";
+
+  // The simplest case: extract 100 comments from a video with more than 100 comments
+  @Test
+  public void testDefaultBehaviour() {
+    List<CommentThread> comments =
+        YouTubeCommentRetriever.retrieveComments(POPULAR_VIDEO_URL, HUNDRED);
```

Comment:
General comments here: it would be recommended to use Mock for YoutubeCommentRetriever instead of directly calling it. The test will not pass in local in current case.  
Note: not sure about this; could be determined by either Rick or Cis:)

Reply:
What do you mean it won't pass in local? When I run it locally, it passes fine.

Reply:
This is where we want Mockito. Your tests depend on actual youtube videos, and it is not a great idea to have extra dependencies for tests.
The following two scenarios of many could happen that could break your tests:
1. The videos no longer exist.
2. Youtube has an outage, you can't retrieve the actual video.

You don't want things that are not related to your actual code change to break your tests, because they shouldn't :)

Reply:
For the previous comments, it runs locally fine bc the youtube API works fine. If youtube API does not work, we cannot test whether the code is correct or not:(

Reply:
I am confused, are we still using real video ids? Can me mock them out?

Reply:
That's a good point. Maybe the solution we came up with doesn't actually prevent the API from being used in the tests.